### ISSUE TITLE
Circular dependency between text_processor and models

### ISSUE THREAD

**Issue by rrunix** [rrunix/tamco_speech#1]
`        
audio_processor = audio.make_audio_processor(audio_args)
text_processor = text.make_text_processor(text_args)

model = acoustic_models.make_model(audio_processor.features_size(), text_processor.number_of_tokens(), model_args)

text_processor.set_special_chars_pos(blank_pos=model.get_blank_pos(), space_pos=model.get_space_pos())
`

The circular depency arises from the fact that the text processor determines the output size, but the model determines the positions of the blank and empty space (Different loss functions require this values to be placed in different indexes such as tensorflow ctc and baidu warpctc)
